### ISSUE TITLE
Support for Namespace documentation in the C# API retypeapp output

### ISSUE THREAD

**Issue by ApolonTorq** [retypeapp/retype#52]
**Feature Request**
Its extremely useful to have namespace level documentation to provide higher level context for the purpose of the classes within a namespace. With previous C# API docs generators, a static NamespaceDoc class could be defined solely to define the comments for a namespace (https://stackoverflow.com/questions/793210/xml-documentation-for-a-namespace).

However though the NamespaceDoc approach is useful, my preference would be a more free flowing markdown solution to introduce a namespace to a reader. This would allow direct editing in markdown rather than editing markdown within a C# comment. The rendered output would have a page for each namespace node in the documentation tree. Currently a namespace has no content in the retype output web UI and is just to open/close a tree node. Potential approaches for namespace markdown include:

- Allow markdown files for namespaces to be defined in the api folder alongside the index.yml file for the API definitions. The convention would be that the markdown file is named after the namespace e.g. System.Collections.Generic.md. Alternatively the namespace could be defined in the metadata section of the markdown file.
- Allow README.md files to be placed in source code folders. The metadata of the README.md file would need to identify which namespace is being documented.
- Just rely on the existing NamespaceDoc.cs convention. Though compatible with other tools, this approach is inconvenient for editing the namespace documentation which is likely to be more descriptive and longer form than the class/member comments.

**Comment by AndreyChechel** [comment]
Hi @ApolonTorq 

Thanks for the nice suggestion and for providing information about `NamespaceDoc` type approach. We'll certainly try to come up with a process of Namespace documentation that would work with Retype seamlessly.

Meanwhile, it's possible to mimic the API folder structure within the `input` folder (where MD files are stored) and to place a MD file with a namespace description at a corresponding location. For example, consider the following structure:

**retype.json**
```json
{
  "input": "./docs",
  "api": {
    "input": "./path/to/csharp/solution",
    "output": "./api"
  }
}
```

If the solution contains a `Demo.Sample.Program` type, the following output structure can be expected:
```
.retype
│   index.html
│
└───api
    └───demo
        └───sample
            └───program
                    index.html
```

So, in order to document the `Sample` namespace, the `input` folder structure should look like:
```
docs
│   index.md
│
└───api
    └───demo
            sample.md
```

The `sample.md` file will be compiled and emitted as `.retype/api/demo/sample/index.html`.

Hope that helps.

**Comment by ApolonTorq** [comment]
Yep, thanks! That works - in combination with order and label metadata to position and title the namespace docs section.
